plutus-conformance/src/PlutusConformance/Common.hs

```diff
@@ -104,21 +40,156 @@ shownParseError = "parse error"
+                Right dbTm                         ->
+                    Right $ MkRawTxtOrDebruijnTerm txt dbTm
+
+-- | Get the tested value (in `Either T.Text RawTxtOrDebruijnTerm`).
+getTestedValue ::
+    UplcEvaluator
+    -> FilePath
+    -> IO (Either T.Text RawTxtOrDebruijnTerm)
+getTestedValue eval dir = do
+    inputFile <- findByExtension [".uplc"] dir
+    case inputFile of
+        [] -> error $ "Input file missing in " <> dir
+        _:_:_ -> error $ "More than 1 input files in " <> dir
+        [file] -> do
+            input <- T.readFile file
+            case parseTxt input of
+                Left _ -> pure $ Left shownParseError
+                Right p -> do
+                    case eval (void p) of
+                        Nothing                           -> pure $ Left shownEvaluationFailure
+                        Just prog@(UPLC.Program () _version tm) -> do
+                            case UPLC.deBruijnTerm tm of
+                                Left (_ :: UPLC.FreeVariableError) ->
+                                    pure $ Left shownEvaluationFailure
+                                Right dbTm                         ->
+                                    pure $ Right $ MkRawTxtOrDebruijnTerm (display prog) dbTm
+
+-- | The comparison function used for the golden test.
+-- This function checks alpha-equivalence of programs when the output is a program.
+compareAlphaEq ::
+    Either T.Text RawTxtOrDebruijnTerm -- ^ golden value
+    -> Either T.Text RawTxtOrDebruijnTerm -- ^ tested value
+    -> IO (Maybe String)
+    -- ^ If two values are the same, it returns `Nothing`.
+    -- If they are different, it returns an error that will be printed to the user.
+compareAlphaEq (Left expectedTxt) (Left actualTxt) =
+    if actualTxt == expectedTxt
+    then pure Nothing
+    else pure $ Just $
+        "Test failed, the output failed to parse or evaluate: \n"
+        <> T.unpack actualTxt
+compareAlphaEq (Right expected) (Right actual) =
+    if inDebruijnTerm actual == inDebruijnTerm expected
+    then pure Nothing
+    else pure $ Just $
+        "Test failed, the output was successfully parsed and evaluated, but it isn't as expected. "
+        <> "The output program is: \n"
+        <> (display $ raw actual)
+        <> "\n The output program, in Debruijn indexes is: \n"
+        <> (show $ inDebruijnTerm actual)
+        -- the user can look at the .expected file, but they can't see it in Debruijn term
+        <> "\n But the expected result in Debruijn indexes is: \n"
+        <> (show $ inDebruijnTerm expected)
+compareAlphaEq (Right expected) (Left actualTxt) =
+    pure $ Just $ "Test failed, the output failed to parse or evaluate: \n"
+        <> T.unpack actualTxt
+        <> "\n But the expected result in Debruijn indexes is: \n"
+        <> (show $ inDebruijnTerm expected)
+compareAlphaEq (Left txt) (Right actual) =
+    if txt == raw actual then pure Nothing
```

Comment:
This is confusing - why does the test pass if the expected program fails to parse? Please add some comments.

Reply:
will do.